File: split/alignpeek.pas
procedure alignpeek;
label
  20;
begin
  20:
    alignstate := 1000000;
  {406:}
repeat
    getxtoken;
  until curcmd <> 10{:406}
;
  if curcmd = 34 then
  begin
    scanleftbrace;
    newsavelevel(7);
    if curlist.modefield = -1 then
      normalparagraph;
  end
  else if curcmd = 2 then
    finalign
  else if (curcmd = 5) and (curchr = 258) then
    goto 20
  else
  begin
    initrow;
    initcol;
  end;
end;

File: cpp/alignpeek.cpp
#include "alignpeek.h"
#include "getxtoken.h"
#include "scanleftbrace.h"
#include "newsavelevel.h"
#include "normalparagraph.h"
#include "finalign.h"
#include "initrow.h"
#include "initcol.h"

void alignpeek(void)
{
	while (true)
	{
		alignstate = 1000000;
		do
			getxtoken();
		while (curcmd == 10);
		if (curcmd == 34)
		{
			scanleftbrace();
			newsavelevel(7);
			if (curlist.modefield == -1)
				normalparagraph();
		}
		else 
			if (curcmd == 2)
				finalign();
			else 
				if (curcmd == 5 && curchr == 258)
					continue;
				else
				{
					initrow();
					initcol();
				}
		break;
	}
}

